pkg/kubelet/container/helpers.go

```diff
@@ -113,6 +113,18 @@ func HashContainer(container *v1.Container) uint64 {
 	// Omit nil or empty field when calculating hash value
 	// Please see https://github.com/kubernetes/kubernetes/issues/53644
 	containerJSON, _ := json.Marshal(container)
+	// container.ResizePolicy is set or defaulted if InPlacePodVerticalScaling feature gate is
+	// enabled. This field is immutable by the user for now. But it can change due to defaulting
+	// kicking in or turning off when this feature gate is toggled in the apiserver regardless of
+	// whether the feature gate is enabled in kubelet. This field is used by kubelet to decide how
+	// to handle pod resize, the usage is limited to kubelet, and does not require container restart.
+	// So, if present, always exclude it from hash.
+	// TODO(vinaykul,InPlacePodVerticalScaling): Always exclude ResizePolicy & Resources from hash on GA+1.
+	if container.ResizePolicy != nil {
+		containerCopy := container.DeepCopy()
+		containerCopy.ResizePolicy = nil
+		containerJSON, _ = json.Marshal(containerCopy)
+	}
```

Comment:
we've had an issue open to fix the approach kubelet takes for determining the need for container restart for 5 years - https://github.com/kubernetes/kubernetes/issues/63814

I *really* think we should fix this correctly by making the kubelet opt *into* fields it knows should trigger container restart, rather than opt *out* of problematic fields like this.

Reply:
The advantage of kubelet fix could also be that it could be back ported to 1.27.X.

Reply:
what is described in https://github.com/kubernetes/kubernetes/issues/63814#issuecomment-760319100 could only be done on a minor boundary when containers are required to be drained / restarted on a given node anyway